geom/src/polygon.rs

```diff
@@ -152,17 +129,26 @@ impl Polygon {
         self.translate(dx, dy)
     }
 
-    // TODO Should be &Ring
-    pub fn get_outer_ring(&self) -> Ring {
-        self.get_rings().remove(0)
+    pub fn get_outer_ring(&self) -> &Ring {
+        &self.rings[0]
     }
 
+    pub fn into_outer_ring(mut self) -> Ring {
+        self.rings.remove(0)
+    }
+
+    /// The [centroid](https://docs.rs/geo/latest/geo/algorithm/centroid/trait.Centroid.html)
```

Comment:
I didn't carefully check through all the callers of this. I don't think any of them care that much about exactly what "center" means.

But also, geo's algorithm can fail. Skimming through the source, I'm not totally sure if it's possible for the polygons we send in here -- it's stuff like the holes taking up the entire polygon.

Reply:
>  it's stuff like the holes taking up the entire polygon.

I think in this case actually, centroid calculation will succeed, it just proceeds with a zero-area centroid though - e.g. treating the polygons as linestrings.

IIRC the only time we'll return a `None` centroid is for empty geometries.

Reply:
Hm, looking at your current method a bit, I'd expect this change to be a potentially serious perf degradation if it's on any hot paths. You were previously taking the centroid of the points, whereas geo's implementation considers area which is relatively a lot more expensive to compute.

I mean, it's still "fast" for what it does, it's just solving a more complicated problem than what you were doing before. If what you had before was working for your use case you might want to revert this portion.